_includes/display-logo.html

```diff
@@ -0,0 +1,66 @@
+{% comment %}
+Logo and screenshots handler.
+{% endcomment %}
+
+{% assign itemLogo = include.itemLogo %}
+{% assign screenshots = include.screenshots %}
+{% assign id = include.id %}
+{% assign classes = include.classes %}
+{% assign codiceIPA = include.codiceIPA | default: "false" %}
+
+{% if itemLogo contains 'github' and itemLogo contains '.svg' %}
+  {% assign logo=itemLogo | append: '?sanitize=true' %}
+{% else %}
+  {% assign logo=itemLogo %}
+{% endif %}
+
+{% if codiceIPA != "false" %}
+{% assign placeholderScreen = "/assets/images/placeholder-software-riuso.jpg" %}
+{% else %}
+{% assign placeholderScreen = "/assets/images/cover_software_opensource.png" %}
+{% endif %}
+
+{% assign picture = screenshots | first | default: logo | default: placeholderScreen %}
+
+{% comment %}
+Show images: 
+1. Screenshot
+2. Software ID
+2. Logo
+3. Default placeholder
+4. If everything else fails, defaults to placeholder (second "url")
+{% endcomment %}
+<div class="sw-{{id}}"></div>
+<script>
+  // these functions are in charge to check
+  // whether software logo exists using
+  // onerror event for img tag.
+  // If it is not exists or other kind of error
+  // are returned from request it will replace
+  // with a simple placeholder.
+  function checkImage (src, good, bad) {
+      var img = new Image();
+      img.onload = good; 
+      img.onerror = bad;
+      img.src = src;
+  }
+
+  function setImage(src, rootEl) {
+    const el = `<div class="{{ classes }}"
+      style="
+        background-position: top center;
+        background-repeat: no-repeat; 
+        background-size: cover;
+        height:150px; ">
+    </div>`;
+    $el = $(el);
+    $el.css('background-image', 'url(' + src + ')');
+    $(rootEl).html($el);
+  }
```

Comment:
These functions are declarated inside this include block file so they will every time this file is required. Imagine a loop of images (very common pattern here) these steps are repeated every time and this is bad. We should then consider to move them in a different place that is common to pages where we use this include.
cc @libremente

Reply:
moved functions in dedicate js file.